dv/uvm/core_ibex/sim.py

```diff
@@ -1,301 +1,529 @@
-          check_ibex_uvm_log(uvm_log, "ibex", test_name, report)
+_CORE_IBEX = os.path.normpath(os.path.join(os.path.dirname(__file__)))
+_IBEX_ROOT = os.path.normpath(os.path.join(_CORE_IBEX, '../../..'))
+_RISCV_DV_ROOT = os.path.join(_IBEX_ROOT, 'vendor/google_riscv-dv')
+_OLD_SYS_PATH = sys.path
+
+# Import riscv_trace_csv and lib from _DV_SCRIPTS before putting sys.path back
+# as it started.
+try:
+    sys.path = ([os.path.join(_CORE_IBEX, 'riscv_dv_extension'),
+                 os.path.join(_RISCV_DV_ROOT, 'scripts')] +
+                sys.path)
+
+    from lib import (process_regression_list,
+                     read_yaml, run_cmd, run_parallel_cmd,
+                     setup_logging, RET_SUCCESS, RET_FAIL)
+    import logging
+
+    from spike_log_to_trace_csv import process_spike_sim_log
+    from ovpsim_log_to_trace_csv import process_ovpsim_sim_log
+    from instr_trace_compare import compare_trace_csv
+
+    from ibex_log_to_trace_csv import process_ibex_sim_log, check_ibex_uvm_log
+
+finally:
+    sys.path = _OLD_SYS_PATH
+
+
+def subst_opt(string, name, enable, replacement):
+    '''Substitute the <name> option in string
+
+    If enable is False, <name> is replaced by '' in string. If it is True,
+    <name> is replaced by replacement, which should be a string or None. If
+    replacement is None and <name> occurs in string, we throw an error.
+
+    '''
+    needle = '<{}>'.format(name)
+    if not enable:
+        return string.replace(needle, '')
+
+    if replacement is None:
+        if needle in string:
+            raise RuntimeError('No replacement defined for {} '
+                               '(used in string: {!r}).'
+                               .format(needle, string))
+        return string
+
+    return string.replace(needle, replacement)
+
+
+def subst_env_vars(string, env_vars):
+    '''Substitute environment variables in string
+
+    env_vars should be a string with a comma-separated list of environment
+    variables to substitute. For each environment variable, V, in the list, any
+    occurrence of <V> in string will be replaced by the value of the
+    environment variable with that name. If <V> occurs in the string but $V is
+    not set in the environment, an error is raised.
+
+    '''
+    env_vars = env_vars.strip()
+    if not env_vars:
+        return string
+
+    for env_var in env_vars.split(','):
+        env_var = env_var.strip()
+        needle = '<{}>'.format(env_var)
+        if needle in string:
+            value = os.environ.get(env_var)
+            if value is None:
+                raise RuntimeError('Cannot substitute {} in command because '
+                                   'the environment variable ${} is not set.'
+                                   .format(needle, env_var))
+            string = string.replace(needle, value)
+
+    return string
+
+
+def subst_cmd(cmd, enable_dict, opts_dict, env_vars):
+    '''Substitute options and environment variables in cmd
```

Comment:
I think you should put the description for argument list

Reply:
I'm sorry, I don't understand your message. I believe that every argument is (quite carefully) documented in this function's docstring.

Reply:
I mean it has description for arguments (Ex cmd, enable_dict, opts_dict, env_vars). For example:
https://github.com/lowRISC/ibex/blob/master/vendor/google_riscv-dv/run.py

Reply:
Do you believe that the three following paragraphs of the docstring fail to document one of the arguments to the function? In which case, please could you say which one?

Reply:
@danghai, are you referring to the format of the comment docstrings?

Reply:
@udinator yes

Reply:
@danghai - I will open a PR in the near future to update the docstring formatting.